### PR TITLE
Update currentDeployed value to only add the current step 

### PR DESCRIPTION
**Pull request by linyguo**

### CODE REVIEW

File: api/pkg/apis/v1alpha1/managers/solution/solution-manager.go [hunk 1]
```diff
@@ -543,7 +543,13 @@ func (s *SolutionManager) Reconcile(ctx context.Context, deployment model.Deploy
 			for _, v := range targetResult {
 				successCount += v
 			}
-			summary.CurrentDeployed += successCount
+			deployedCount := 0
+			for _, ret := range componentResults {
+				if ret.Status == v1alpha2.Updated { // TODO: need to ensure the status updated correctly on returning from target providers.
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/managers/solution/solution-manager.go:548]: Btw, if it is delete case, will the expected return status is also Updated?

Comment (linyguo) [reply api/pkg/apis/v1alpha1/managers/solution/solution-manager.go]: it should return "Deleted" but some providers are still returning "Untouched". I updated the status check here and will add this to issue #513

File: api/pkg/apis/v1alpha1/managers/solution/solution-manager.go [hunk 2]
```diff
@@ -484,6 +484,7 @@ func (s *SolutionManager) Reconcile(ctx context.Context, deployment model.Deploy
 				log.InfofCtx(ctx, " M (Solution): skipping step with role %s on target %s", step.Role, step.Target)
 				targetResult[step.Target] = 1
 				planSuccessCount++
+				summary.CurrentDeployed += len(step.Components)
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/managers/solution/solution-manager.go:487]: Why we count the currentDeployed here? Deployment hasn't started yet in this place.

Comment (linyguo) [reply api/pkg/apis/v1alpha1/managers/solution/solution-manager.go:487]: add the components for skipped step as it will move on to next step

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**